cpp/include/cudf/detail/utilities/alignment.hpp

```diff
@@ -42,5 +42,17 @@ T* align_ptr_for_type(void* destination)
     std::align(alignment, bytes_needed, destination, padded_bytes_needed));
 }
 
+/**
+ * @brief Align a value up to the nearest multiple of the given alignment.
+ *
+ * @param value The input value to align
+ * @param alignment The amount of bytes to align, must be a power of 2
+ * @return The aligned value
+ */
+inline std::size_t align_up(std::size_t value, std::size_t alignment)
```

Comment:
Is this the same as `round_up_safe`? https://github.com/rapidsai/cudf/blob/54b81df4e189cf36aca306a73b93dd68a717a80c/cpp/include/cudf/detail/utilities/integer_utils.hpp#L47

Reply:
I think this function could be useful if it took the number of alignment bits. Removes the power of 2 requirement and differentiates it from `round_up_safe`.

Reply:
Sorry I'm confused. What is the conclusion here? Should `round_up_safe` be used instead of adding `align_up`?

Reply:
I would prefer just using `round_up_safe` here and using a `constexpr` named variable for the round-up value instead of the  hardcoded 8 value.

Reply:
Indeed, `round_up_safe` in this use case produces exactly the same output as `align_up`. So I'm going to switch to use it, as suggested by @davidwendt to get the bugs fixed ASAP. `padded_device_buffer` will be the follow up work.

Reply:
Could have also used RMM's function?

Reply:
Finished adoption of `round_up_safe` but still strugging to find a good place to put in the `constexpr` that @davidwendt recommended.

Reply:
The changes seem to correspond to temporary compression/decompression buffers. Perhaps `gpuinflate.hpp`? 
I'm ok defining it in each .cu file too if there is a chance the padding may be different for the different implementations.

Reply:
> Could have also used RMM's function?

I  think we can. However, that is a detail function (`rmm::detail::align_up`) so I was reluctant to use it.